src/core/common/instance.hpp

```diff
@@ -312,17 +312,23 @@ class Instance : public otInstance
     TimerMicroScheduler mTimerMicroScheduler;
 #endif
 
-#if OPENTHREAD_MTD || OPENTHREAD_FTD
     // RandomManager is initialized before other objects. Note that it
     // requires MbedTls which itself may use Heap.
-#if !OPENTHREAD_CONFIG_MULTIPLE_INSTANCE_ENABLE
+
+#if OPENTHREAD_MTD || OPENTHREAD_FTD
+#if !OPENTHREAD_ENABLE_MULTIPLE_INSTANCES
```

Comment:
Is this intended to regress the recent move to `OPENTHREAD_CONFIG_MULTIPLE_INSTANCE_ENABLE` above?

Reply:
The diff here is a bit hard to read. The change is actually just moving the comment above the `#if` checks instead of in middle of them.